### PR TITLE
Cleaning up Python module import infrastructure

### PR DESCRIPTION
**Pull request by jcotela**
Now that #3586 is merged, the Python import infrastructure can be greatly simplified. This PR cleans up some of the darker things going on when an application is loaded. We have also taken the occasion to finally remove the old `application_importer`, which has been deprecated for 7 years now...

This PR is 100% backwards compatible but it also introduces a "more Pythonic" module import function `_ImportApplicationAsModule`. This one is **optional** but, if application developers choose to use it in their application's `__init__` instead the current one, it **does not** add the application#s `python_scripts` folder to the python path. This means that python scripts would have to be imported relative to the application
```Python
import KratosMultiphysics
import KratosMultiphysics.FluidDynamicsApplication as kfd

# the next one is the big change 
import kfd.navier_stokes_solver_vmsmonolithic as solver
# alternatively
from kfd.navier_stokes_solver_vmsmonolithic import NavierStokesSolverMonolithic
```

This allows Kratos applications to work like native python submodules but, since it would break current scripts, it is not enabled by default.

### CODE REVIEW

File: applications/MappingApplication/python_scripts/__init__.py [hunk 1]
```diff
@@ -2,13 +2,10 @@
...
 application = KratosMappingApplication()
 application_name = "KratosMappingApplication"
 application_folder = "MappingApplication"
 
-# The following lines are common for all applications
-from . import application_importer
-import inspect
-caller = inspect.stack()[1] # Information about the file that imported this, to check for unexpected imports
-application_importer.ImportApplication(application,application_name,application_folder,caller)
+KM._ImportApplicationAsModule(application, application_name, application_folder, __path__)
```
Comment (philbucher) [inline applications/MappingApplication/python_scripts/__init__.py]: note that we ported the MappingApp right away to see how it is being used

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee it would be nice to have this asap
Two rather large apps (IGA and CoSimulation) could then use it from the beginning and users would not have to update painfully at some point

**Review by pooyan-dadvand — APPROVED** [review]
@KratosMultiphysics/technical-committee approves the cleanup but let put on hold the optional part to maintain the backward compatibility

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee just to clarify:
The change introduced here only removes the "old" way of adding stuff to the path for the next-gen python-interface of Kratos
Right now with the change I introduced in #3217 we have two ways of using python-modules: the pythonic way (e.g. `import KratosMultiphysics.process_factory`) or the "old"direct way (e.g. `import process_factory`)

The big apps already now **support both ways** (the later one for backwards compatibility)

Only new apps will use right away the new/correct/pythonic import system. There we don't need backwards-compatibility bcs there is no "old" way

From a user-point of view we already now use the "new" way for old and new apps
=> see e.g. the `MainKratos.py` in GiD: 
[FluidMainKratos](https://github.com/KratosMultiphysics/GiDInterface/blob/master/kratos.gid/apps/Fluid/python/KratosFluid.py#L4)
[StructuralMainKratos](https://github.com/KratosMultiphysics/GiDInterface/blob/master/kratos.gid/apps/Structural/python/KratosStructural.py#L4)

**Pull request merged**